src/Core/Registration/GlobalOptimization.cpp

```diff
@@ -535,8 +537,25 @@ void GlobalOptimizationLevenbergMarquardt::
         timer_iter.Start();
         int lm_count = 0;
         do {
-            Eigen::MatrixXd H_LM = H + current_lambda * H_I;
-            Eigen::VectorXd delta = H_LM.ldlt().solve(b);
+			Eigen::MatrixXd H_LM = H + current_lambda * H_I;
+			Eigen::VectorXd delta(H_LM.cols());
+
+			//Using a sparse solver
+			Eigen::SparseMatrix<double> H_LM_sparse = H_LM.sparseView();
+			Eigen::SimplicialCholesky<Eigen::SparseMatrix<double>> chol;
```

Comment:
Do you know under the hood it is ``SimplicialLLt`` or ``SimplicialLDLt``?

Reply:
A cursory glance at the ``SimplicialCholesky.h`` file reveals that ``SimpliciallCholesky`` uses ``LDLT`` by default unless otherwise specified.

```C++
 template<typename _MatrixType, int _UpLo, typename _Ordering>
     class SimplicialCholesky : public SimplicialCholeskyBase<SimplicialCholesky<_MatrixType,_UpLo,_Ordering> >
 {
 public:
     typedef _MatrixType MatrixType;
     enum { UpLo = _UpLo };
     typedef SimplicialCholeskyBase<SimplicialCholesky> Base;
     typedef typename MatrixType::Scalar Scalar;
     typedef typename MatrixType::RealScalar RealScalar;
     typedef typename MatrixType::StorageIndex StorageIndex;
     typedef SparseMatrix<Scalar,ColMajor,StorageIndex> CholMatrixType;
     typedef Matrix<Scalar,Dynamic,1> VectorType;
     typedef internal::traits<SimplicialCholesky> Traits;
     typedef internal::traits<SimplicialLDLT<MatrixType,UpLo> > LDLTTraits;
     typedef internal::traits<SimplicialLLT<MatrixType,UpLo>  > LLTTraits;
   public:
     SimplicialCholesky() : Base(), m_LDLT(true) {}
 
     explicit SimplicialCholesky(const MatrixType& matrix)
       : Base(), m_LDLT(true)
     {
       compute(matrix);
     }
 
     SimplicialCholesky& setMode(SimplicialCholeskyMode mode)
     {
       switch(mode)
       {
       case SimplicialCholeskyLLT:
         m_LDLT = false;
         break;
       case SimplicialCholeskyLDLT:
         m_LDLT = true;
         break;
       default:
         break;
       }
 
       return *this;
  }
```